broker/utils.go

```diff
@@ -10,19 +10,18 @@ import (
 	composeapi "github.com/compose/gocomposeapi"
 )
 
+var errDeploymentNotFound = errors.New("Deployment not found")
+
 func findDeployment(c compose.Client, name string) (*composeapi.Deployment, error) {
```

Comment:
The point of this function is to abstract away some of the details of the compose client. This also encapsulates the error handling including the somewhat ugly string matching on the returned error. See the commit "Don't return ErrInstanceDoesNotExist for all errors" for more details.

TBH I'm not 100% happy with the upstream lib's error handling. For one thing, they return a slice of errors, and I'm not clear why that it. It's not very idiomatic Go. Secondly, it would be much better if they returned some concrete errors so that we can tell the difference between a not-found error and other errors without having to resort to string matching. I might raise a PR against upstream for this if I get a moment.

In the meantime, I think this function is necessary to encapsulate these details.